docs/plugins/gatsby-remark-relative-linker/functions.js

```diff
@@ -0,0 +1,60 @@
+const SIBLING = "./";
+const PARENT = "../";
+const UNPREFIXED = -1;
+
+function* depthCheck(n) {
+  for (let i = n; i > -1; i--) yield i;
+}
+
+function escapeRegExp(string) {
+  //https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping
+  return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+}
+
+function buildDepthCheckRegex(depth) {
+  if (depth === 0) return new RegExp("^" + escapeRegExp(SIBLING));
+  return new RegExp("^" + escapeRegExp(PARENT.repeat(depth)));
+}
+
+function doUpdateDepth(linkText, forDepth, isIndex = false) {
+  let noPrefixRegex = /^[\w-_]+/;
+
+  if (linkText.match(noPrefixRegex)) {
+    return updateLinkText(linkText, UNPREFIXED, isIndex);
+  }
+
+  let depthCheckRegex = buildDepthCheckRegex(forDepth);
+  let result = linkText.match(depthCheckRegex);
+
+  if (result) {
+    return updateLinkText(linkText, forDepth, isIndex);
+  }
+}
+
+function updateLinkText(linkText, depth, index) {
+  switch (index) {
+    case true:
+      return depth === UNPREFIXED ? SIBLING + linkText : linkText;
+    case false:
+      return depth === UNPREFIXED
+        ? PARENT + linkText
+        : depth === 0
+        ? linkText.replace(SIBLING, PARENT)
+        : linkText.replace(
+            buildDepthCheckRegex(depth),
+            PARENT.repeat(depth + 1)
+          );
```

Comment:
it might be just me but I find the nested ? : syntax quite not straightforward to follow. Originally I was thinking this function can be simplified to be: 

function updateLinkText(linkText, depth, index) {
  switch (index) {
    case true:
      return linkText;
    case false:
      return "../" + linkText;
}

Another reason behind prefixing ../ directly is: 
what if developer writes language/../../docs/introduction.md -- would this work with current logic? I think this change will not process this case as it does not begin with ../../ ?